gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/util/DefaultGraphManager.java

```diff
@@ -0,0 +1,210 @@
+     *         {@link TraversalSource} itself
+     */
+    public final Map<String, TraversalSource> getTraversalSources() {
+        return traversalSources;
+    }
+
+    public final TraversalSource getTraversalSource(final String tsName) {
+        return traversalSources.get(tsName);
+    }
+
+    public final void addTraversalSource(final String tsName, final TraversalSource ts) {
+        traversalSources.put(tsName, ts);
+    }
+
+    /**
+     * Get the {@link Graph} and {@link TraversalSource} list as a set of bindings.
+     */
+    public final Bindings getAsBindings() {
+        final Bindings bindings = new SimpleBindings();
+        graphs.forEach(bindings::put);
+        traversalSources.forEach(bindings::put);
+        return bindings;
+    }
+
+    /**
+     * Rollback transactions across all {@link Graph} objects.
+     */
+    public final void rollbackAll() {
+        graphs.entrySet().forEach(e -> {
+            final Graph graph = e.getValue();
+            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
+                graph.tx().rollback();
+        });
+    }
+
+    /**
+     * Selectively rollback transactions on the specified graphs or the graphs of traversal sources.
+     */
+    public final void rollback(final Set<String> graphSourceNamesToCloseTxOn) {
+        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.ROLLBACK);
+    }
+
+    /**
+     * Commit transactions across all {@link Graph} objects.
+     */
+    public final void commitAll() {
+        graphs.entrySet().forEach(e -> {
+            final Graph graph = e.getValue();
+            if (graph.features().graph().supportsTransactions() && graph.tx().isOpen())
+                graph.tx().commit();
+        });
+    }
+
+    /**
+     * Selectively commit transactions on the specified graphs or the graphs of traversal sources.
+     */
+    public final void commit(final Set<String> graphSourceNamesToCloseTxOn) {
+        closeTx(graphSourceNamesToCloseTxOn, Transaction.Status.COMMIT);
+    }
+
+    /**
+     * If {@link Map} containing {@link Graph} references contains one corresponding to
+     * {@link String} graphName, then we return that {@link Graph}; otherwise, instantiate a
+     * new {@link Graph} using the {@link Supplier}, add it to the {@link Map} tracking {@link Graph}
+     * references, and return that {@link Graph}. 
+     */
+    public final Graph openGraph(final String graphName, final Supplier<Graph> supplier) {
+        final Graph graph = graphs.get(graphName);
+        if (null != graph) {
+            return graph;
+        }
+        final Graph newGraph = supplier.get();
+        addGraph(graphName, newGraph);
+        return newGraph;
+    }
+
+    /**
+     * Remove {@link Graph} corresponding to {@link String} graphName from {@link Map}
+     * tracking graph references.
+     */
+    public final void removeGraph(final String graphName) {
```

Comment:
Looking at this signature here, I think it would be better to mimic `Map.remove(k)` and return `Graph` as in:

```java
public final Graph removeGraph(final String graphName) 
```

Then a caller would then just do:

```java
manager.removeGraph("graph").close();
```

or whatever they would do with their dynamically added graph.